frontend/src/Frontend/UI/Dialogs/WatchRequest.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ExtendedDefaultRules #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE RecursiveDo #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
+
+-- | Dialog for watching request keys
+-- Copyright   :  (C) 2020 Kadena
+-- License     :  BSD-style (see the file LICENSE)
+module Frontend.UI.Dialogs.WatchRequest
+  ( uiWatchRequestDialog
+  ) where
+
+import Control.Applicative (liftA2)
+import Control.Lens hiding (failover)
+import Control.Monad (void)
+import Data.Either (rights)
+import Reflex.Dom
+import Safe (headMay)
+import qualified Pact.Types.Command as Pact
+import qualified Pact.Types.Util as Pact
+
+import Frontend.Foundation hiding (Arg)
+import Frontend.Network
+import Frontend.UI.Dialogs.DeployConfirmation (Status(..), listenToRequestKey, transactionStatusSection, transactionResultSection)
+import Frontend.UI.Modal
+import Frontend.UI.Widgets
+import Frontend.UI.Widgets.Helpers (dialogSectionHeading)
+
+-- | A modal for watching request keys
+uiWatchRequestDialog
+  :: ( Flattenable mConf t
+     , Monoid mConf
+     , HasNetwork model t
+     , MonadWidget t m
+     )
+  => model -> Event t () -> m (mConf, Event t ())
+uiWatchRequestDialog model onCloseExternal = do
+  (conf, closes) <- fmap splitDynPure $ workflow $ inputRequestKey model onCloseExternal
+  mConf <- flatten =<< tagOnPostBuild conf
+  let close = switch $ current closes
+  pure (mConf, close)
+
+-- | Allow the user to input a request key and chain
+inputRequestKey
+  :: ( Monoid mConf
+     , MonadWidget t m
+     , HasNetwork model t
+     )
+  => model
+  -> Event t () -- ^ Modal was externally closed
+  -> Workflow t m (mConf, Event t ())
+inputRequestKey model onCloseExternal = Workflow $ mdo
+  let nodes = fmap rights $ model ^. network_selectedNodes
+  close <- modalHeader $ text "Watch Request Key"
+  choice <- modalMain $ do
+    dialogSectionHeading mempty "Notice"
+    divClass "group" $ do
+      text "If you have a request key from a previously submitted transaction, you can use this dialog to wait for and display the results."
+    dialogSectionHeading mempty "Required Information"
+    info <- divClass "group" $ do
+      mChain <- mkLabeledClsInput False "Chain ID" $ uiChainSelection (headMay <$> nodes) (pure Nothing)
+      eKey <- mkLabeledInput False "Request Key" uiInputElement def
+      let parseRequestKey t
+            | {- T.length t == 43, -} Right v <- Pact.fromText' t = Right v
```

Comment:
eh?

Reply:
All the hashes appear to be 43 characters long. It seemed weird using the dialog without this, because you could pretty much enter anything you wanted and progress to the next screen. I'm not sure if these hashes will always be that long, though, or if the few times I did transactions they just happened to be that long. Removed the comment anyway, I forgot I'd left it in.